streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -100,7 +105,8 @@ public void run() {
             } catch (final RuntimeException anyOtherException) {
                 handleRuntimeException(anyOtherException);
             } finally {
-                clear();
+                removeAddedTasksFromInputQueue();
+                removeUpdatingAndPausedTasks();
```

Comment:
We no longer clear the `restoredActiveTasks`, is that intentional?

Do we assume by the time the thread is shutting down, the stream thread would not care about any restored tasks any longer, or are you going to merge the restored tasks with removed tasks soon anyways?

Reply:
Yeah, this is intentional. We need to close the task that are in the the `restoredActiveTasks` queue. If I cleared that queue here, we would lose the references to those tasks and we would not be able to ever close them. I have an additional PR that moves the tasks from the output queues to the task registry where they can be closed during the shutdown of the task manager.